### PR TITLE
 Legrand Bticino Living Light L4027C

### PR DESCRIPTION
**Pull request by Smanar**
- Improve support of Legrand Bticino Living Light L4027C (Shutter SW with level control)
  Handle inverted lift values.

- Improve Legrand covering support  https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2812
Some device have different firmware support.

### CODE REVIEW

File: rest_lights.cpp [hunk 1]
```diff
@@ -1824,6 +1834,24 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
...
+                 (taskRef.lightNode->modelId() == QLatin1String("Shutter SW with level control")) )
+            {
+                taskRef.lightNode->setValue(RStateLift, 50);
+                taskRef.lightNode->setValue(RStateBri, 127);
+                //taskRef.lightNode->setValue(RStateLift, targetLift);
+                //taskRef.lightNode->setValue(RStateBri, targetLift * 254 / 100);
+                //bool on2 = targetLift > 0;
+                //bool open2 = targetLift < 100;
+                //taskRef.lightNode->setValue(RStateOpen, open2);
+                //taskRef.lightNode->setValue(RStateOn, on2);
```
Comment (manup) [inline rest_lights.cpp:1853]: Please remove commented out dead code, usually it doesn't age well :)

File: rest_lights.cpp [hunk 2]
```diff
@@ -1862,6 +1890,21 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
...
+            // Because the attribute reporting take realy long time to be done, can be 2 minutes
+            // Or it can be changed only after this time, so using an read attribute don't give usable value
+            // And can cause issue on some third app
+            if ((taskRef.lightNode->modelId() == QLatin1String("Shutter switch with neutral")) ||
+                 (taskRef.lightNode->modelId() == QLatin1String("Shutter SW with level control")) )
+            {
+                //taskRef.lightNode->setValue(RStateOpen, targetOpen);
+                //taskRef.lightNode->setValue(RStateOn, !targetOpen);
+                //taskRef.lightNode->setValue(RStateBri, targetOpen ? 0 : 255);
+                //taskRef.lightNode->setValue(RStateLift, targetOpen ? 0 : 100);
```
Comment (manup) [inline rest_lights.cpp:1904]: Dead code

**Review by manup — CHANGES_REQUESTED** [review]

File: rest_lights.cpp (hunk 1 above)
Comment (Smanar) [reply rest_lights.cpp]: done

File: rest_lights.cpp (hunk 2 above)
Comment (Smanar) [reply rest_lights.cpp]: And done again, was to switch easily from 1 method to another to make tests.

**Review by manup — APPROVED** [review]

**Pull request merged**